src/test/java/api/support/fakes/storage/Storage.java

```diff
@@ -0,0 +1,35 @@
+package api.support.fakes.storage;
+
+import static api.support.APITestContext.getTenantId;
+
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+import io.vertx.core.json.JsonObject;
+
+public final class Storage {
+  private static final Storage storage = new Storage();
+
+  private final Map<String, Map<String, JsonObject>> resources = new HashMap<>();
+
+  public static Storage getStorage() {
+    return storage;
+  }
+
+  public Map<String, JsonObject> getTenantResources(String rootPath, String tenant) {
+    return resources.computeIfAbsent(getKey(rootPath, tenant), k -> new HashMap<>());
+  }
+
+  public Map<String, JsonObject> getTenantResources(URL wholePath) {
+    return getTenantResources(wholePath.getPath(), getTenantId());
```

Comment:
How does this work for a path like `/circulation/loans/123456`, won't this create a separate resource for every record?

Reply:
Will remove this method. It was created to simplify logic in pre-processors. The fake storage module is not using this method.